openpype/pipeline/workfile/workfile_doc.py

```diff
@@ -72,18 +72,18 @@ def set_workfile_note(project_name, asset_id, task_name, filepath, note):
         asset_id (ObjectID): Asset ID.
         task_name (str): Task name.
         filepath (str): Workfile filepath.
-        note (str): Artist note content.
+        data (dict): Workfile data content.
     """
     # This does not create the document if it already exists.
     workfile_doc = create_workfile_doc(
         project_name, asset_id, task_name, filepath
     )
-    if workfile_doc.get("data", {}).get("note") == note:
+    if workfile_doc.get("data") == data:
         # Nothing to update
         return
 
     new_workfile_doc = copy.deepcopy(workfile_doc)
-    new_workfile_doc.setdefault("data", {})["note"] = note
+    new_workfile_doc["data"] = data
```

Comment:
This will completely replace `data` - which I'm not sure is what we'd always want. 

Calling `set_workfile_data(... , {"note": "y"}` will essentially now remove all other data? Seems like we'd want more selective updates? How about `new_workfile_doc["data"].update(data)`?